File: frontend/src/DataClasses/UserInfo.ts
import Token from "./Token";
import UserType from "./UserType";

interface UserInfo {
    oid: string;
    name: string;
    surname: string;
    email: string;
    accountType: UserType;
    isBanned: boolean;
}

export const getUserInfoFromToken = (token: Token): UserInfo => ({
    oid: token.oid,
    name: token.name,
    surname: token.surname,
    email: token.email,
    accountType: token.userType,
    isBanned: token.isBanned
});

export default UserInfo;

File: backend/src/Infrastructure/Identity/UserInfo.cs
namespace PartyKlinest.Infrastructure.Identity
{
    public class UserInfo
    {
        public string OID { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string Email { get; set; }
        public string AccountType { get; set; }
        public bool IsBanned { get; set; }

        public UserInfo(Token token)
        {
            OID = token.OID;
            Name = token.Name;
            Surname = token.Surname;
            Email = token.Email;
            AccountType = UserTypeConverter.EnumToString(token.UserType);
            IsBanned = token.IsBanned;
        }

        public UserInfo(string oID, string name, string surname, string email, string accountType, bool isBanned)
        {
            OID = oID;
            Name = name;
            Surname = surname;
            Email = email;
            AccountType = accountType;
            IsBanned = isBanned;
        }
    }
}
